### PR TITLE
Implementation plan  for security baseline SIG onboarding

### PR DESCRIPTION
**Pull request by Danajoyluck**
Detailed implementation plan for onboarding security baseline SIG.

### CODE REVIEW

File: operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## Summary of Changes
+The plan is to: 
+* Create a new GitHub repo in “ossf” org
+* Create GitHub Teams for the new repo access
+* Establish members in the teams 
+* Validate the above artifacts are created successfully
+
+## Prerequisites
+* [Prerequist](Prerequisites) in the onboadring radme.md
```
Comment (david-a-wheeler) [inline operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md:16]: Spelling nit: "onboadring radme.md" should be "onboarding readme.md"

File: operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Implementation Time
+As soon as this plan is approved and merged to main.
+
+## Implementation
+[security_baseline.json](https://github.com/ossf/staff/blob/main/operations/github/onboarding/repo_init_metadata/security_baseline.json) contains all the data for the repo setup. 
+1. Get the latest code
+2. Run below command to
+* Create teams, 1 main team  and 2 sub teams with different repo roles
+* Add members to different teams, assign them team roles
+* Creat repo
```
Comment (david-a-wheeler) [inline operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md:33]: Nit: "Creat" -> "Create"

File: operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+* Assign teams' access to the repo
+* Create a test issue
+* Look up all the artigfacts created from above and write the search results to security_baseline_results.json
```
Comment (david-a-wheeler) [inline operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md:36]: artigfacts ?

**Comment by david-a-wheeler** [comment]
While reading I noted some spelling nits (see above), but really, this looks fine.

This is for simply onboarding / creating the infrastructure for the SIG (mainly its repo) itself, and that's fine. Its README will need to make it clear who the leaders are, and that it's part of the Best Practices WG. The WG should also point to this, once it's set up.

An alternative would be to house this *within* the best practices WG. The WG already has a setup to create pretty web pages, while if you want to do that in this SIG, there may be an extra step. However, because there are a number of other foundations involved, I think the approach (as proposed) is the better way anyway.

We may eventually want to create materials that will *implement* the baseline. That work might be part of this repo, or a separate repo. I would suggest putting it all in this repo if that makes sense, and if we discover it doesn't, we'll have a rationale for creating another repo.

Thanks for doing this!

File: operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md (hunk 3 above)
Comment (Danajoyluck) [reply operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md]: checked in update. Thank you @david-a-wheeler.

File: operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md (hunk 2 above)
Comment (Danajoyluck) [reply operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md]: checked in update. Thank you @david-a-wheeler.

File: operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md (hunk 1 above)
Comment (Danajoyluck) [reply operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md]: checked in update. Thank you @david-a-wheeler.

**Review by david-a-wheeler — APPROVED** [review]

File: operations/github/onboarding/repo_init_metadata/security_baseline.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
   },
   "main_team": {
       "name": "sig-security-baseline",
-      "parent_team": "test",
+      "parent_team": "SIGs",
       "privacy": "closed",
       "notification_setting": "true",
       "permission": "pull",
```
Comment (bbpursell1) [inline operations/github/onboarding/repo_init_metadata/security_baseline.json:14]: Is this a valid permission?  I looks like you're using this in the call to this https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#add-or-update-team-project-permissions

That lists only "read, write, admin" as the possible values.  I think the "pull" value is deprecated.  I would assume that this should allow for the "maintain" role, but it isn't in the documentation.

Comment (Danajoyluck) [reply operations/github/onboarding/repo_init_metadata/security_baseline.json:14]: this is the API that the code uses to add or update repo access
https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#add-or-update-team-repository-permissions

**Review by bbpursell1 — APPROVED** [review]
After discussion and clarification on the permissions, this looks good.

**Pull request merged**